lib/homework_screen/widgets/edit_assignment_widget/edit_assignment_sheet.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:flutter/material.dart';
+import 'package:lachies_life_planner/homework_screen/models/assignment.dart';
+import 'package:lachies_life_planner/homework_screen/models/new_assignment.dart';
+import 'package:lachies_life_planner/homework_screen/widgets/edit_assignment_widget/edit_assignment_add_button.dart';
+import 'package:lachies_life_planner/homework_screen/widgets/edit_assignment_widget/edit_assignment_cancel_button.dart';
+import 'package:lachies_life_planner/homework_screen/widgets/edit_assignment_widget/edit_assignment_form.dart';
+import 'package:lachies_life_planner/homework_screen/widgets/edit_assignment_widget/edit_assignment_update_button.dart';
+import 'package:lachies_life_planner/shared/config/size_config.dart';
+import 'package:provider/provider.dart';
+
+class EditAssignmentSheet extends StatefulWidget {
+  final Assignment assignment;
+
+  const EditAssignmentSheet({this.assignment});
+
+  @override
+  _EditAssignmentWidgetState createState() => _EditAssignmentWidgetState();
+}
+
+class _EditAssignmentWidgetState extends State<EditAssignmentSheet> {
+  final GlobalKey<FormState> formKey = GlobalKey<FormState>();
+  TextEditingController _assignmentEditingController;
+
+  @override
+  void initState() {
+    super.initState();
+    _initTextController();
+  }
+
+  @override
+  void dispose() {
+    _assignmentEditingController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Provider<NewAssignment>(
+      create: (BuildContext context) =>
+          NewAssignment.fromAssignment(widget.assignment, formKey) ?? NewAssignment(formKey),
```

Comment:
Would a global formkey resolve this issue?  Why can't I always pass widget.assignment?